tests/core/pyspec/eth2spec/test/altair/sync_protocol/test_light_client_sync.py

```diff
@@ -0,0 +1,424 @@
+from typing import (Any, Dict, List)
+
+from eth_utils import encode_hex
+from eth2spec.test.context import (
+    spec_state_test_with_matching_config,
+    with_presets,
+    with_altair_and_later,
+)
+from eth2spec.test.helpers.attestations import (
+    next_slots_with_attestations,
+    state_transition_with_full_block,
+)
+from eth2spec.test.helpers.constants import MINIMAL
+from eth2spec.test.helpers.light_client import (
+    get_sync_aggregate,
+)
+from eth2spec.test.helpers.state import (
+    next_slots,
+    transition_to,
+)
+
+
+def setup_test(spec, state):
+    class LightClientSyncTest(object):
+        steps: List[Dict[str, Any]]
+        genesis_validators_root: spec.Root
+        store: spec.LightClientStore
+
+    test = LightClientSyncTest()
+    test.steps = []
+
+    yield "genesis_validators_root", "meta", "0x" + state.genesis_validators_root.hex()
+    test.genesis_validators_root = state.genesis_validators_root
+
+    next_slots(spec, state, spec.SLOTS_PER_EPOCH * 2 - 1)
+    trusted_block = state_transition_with_full_block(spec, state, True, True)
+    trusted_block_root = trusted_block.message.hash_tree_root()
+    bootstrap = spec.create_light_client_bootstrap(state)
+    yield "trusted_block_root", "meta", "0x" + trusted_block_root.hex()
+    yield "bootstrap", bootstrap
+    test.store = spec.initialize_light_client_store(trusted_block_root, bootstrap)
+
+    return test
+
+
+def finalize_test(test):
+    yield "steps", test.steps
+    yield "expected_finalized_header", test.store.finalized_header
+    yield "expected_optimistic_header", test.store.optimistic_header
+
+
+def get_update_file_name(spec, update):
+    if spec.is_sync_committee_update(update):
+        suffix1 = "s"
+    else:
+        suffix1 = "x"
+    if spec.is_finality_update(update):
+        suffix2 = "f"
+    else:
+        suffix2 = "x"
+    return f"update_{encode_hex(update.attested_header.hash_tree_root())}_{suffix1}{suffix2}"
+
+
+def emit_slot(test, spec, state):
+    current_slot = state.slot
+    yield from []
+    test.steps.append({
+        "process_slot": {
+            "current_slot": int(current_slot),
+        }
+    })
```

Comment:
To make debugging easier, rather than output `expected_finalized_header` and `expected_optimistic_header`, we can output the current `test.store.finalized_header.hash_tree_root()` and `test.store.optimistic_header.hash_tree_root()` after *each* `process_slot` and `process_update`.

i.e.,

```suggestion
test.steps.append({
        "process_slot": {
            "current_slot": int(current_slot),
        },
        "checks": {
            "finalized_header": {
                'slot': int(test.store.finalized_header.slot),
                'root': encode_hex(test.store.finalized_header.hash_tree_root()),
            },
            "optimistic_header_root": {
                'slot': int(test.store.optimistic_header.slot),
                'root': encode_hex(test.store.optimistic_header.hash_tree_root()),
            }
        },
    })
```

Reply:
3c5d347cdc66e3cb40f248ce1dfe4def4b7c6ad8